tests/comfy_cli/jobs/test_jobs.py

```
@@ -2519,11 +2526,82 @@ def test_watch_executed_collects_output_urls_with_host_port():
     assert st.end_reason == "error"
     assert st.end_details == data
+    # The `executed` list is the only record of what ran before the failure.
+    assert st.completed_nodes == {"1", "2"}
+
+
+def test_watch_progress_state_streams_per_node_and_marks_finished_nodes():
+    """`progress_state` (current ComfyUI) must produce per-step progress events
+    and count finished nodes — including compute nodes that never fire
+    `executed`. Regression: watch only understood the legacy `progress` type."""
+    st, r = _watch_state()
+    jobs_mod._watch_progress_state(
```

Pylint (4.0.6)

[warning] Access to a protected member _watch_progress_state of a client class

(W0212)